src/planner/binder/statement/bind_create_table.cpp

```diff
@@ -142,50 +58,120 @@ vector<unique_ptr<BoundConstraint>> Binder::BindNewConstraints(vector<unique_ptr
 
 	// handle primary keys/not null constraints
 	bool has_primary_key = false;
-	logical_index_set_t not_null_columns;
-	vector<LogicalIndex> primary_keys;
-	for (idx_t c = 0; c < constraints.size(); c++) {
-		auto &constr = constraints[c];
-		switch (constr->type) {
-		case ConstraintType::NOT_NULL: {
-			auto &not_null = constr->Cast<NotNullConstraint>();
-			auto &col = columns.GetColumn(LogicalIndex(not_null.index));
-			bound_constraints.push_back(make_uniq<BoundNotNullConstraint>(PhysicalIndex(col.StorageOid())));
+	physical_index_set_t not_null_columns;
+	vector<PhysicalIndex> primary_keys;
+	for (const auto &bound_constr : bound_constraints) {
+		if (bound_constr->type == ConstraintType::NOT_NULL) {
+			auto &not_null = bound_constr->Cast<BoundNotNullConstraint>();
 			not_null_columns.insert(not_null.index);
-			break;
-		}
-		case ConstraintType::UNIQUE: {
-			auto &unique = constr->Cast<UniqueConstraint>();
-			auto &bound_unique = bound_constraints[c]->Cast<BoundUniqueConstraint>();
-			if (unique.IsPrimaryKey()) {
+		} else if (bound_constr->type == ConstraintType::UNIQUE) {
+			auto &unique = bound_constr->Cast<BoundUniqueConstraint>();
+			if (unique.is_primary_key) {
 				// we can only have one primary key per table
 				if (has_primary_key) {
 					throw ParserException("table \"%s\" has more than one primary key", table_name);
 				}
 				has_primary_key = true;
-				primary_keys = bound_unique.keys;
+				primary_keys = unique.keys;
 			}
-			break;
-		}
-		default:
-			break;
 		}
 	}
+
 	if (has_primary_key) {
 		// if there is a primary key index, also create a NOT NULL constraint for each of the columns
 		for (auto &column_index : primary_keys) {
 			if (not_null_columns.count(column_index)) {
 				//! No need to create a NotNullConstraint, it's already present
 				continue;
 			}
-			auto physical_index = columns.LogicalToPhysical(column_index);
-			constraints.push_back(make_uniq<NotNullConstraint>(column_index));
-			bound_constraints.push_back(make_uniq<BoundNotNullConstraint>(physical_index));
+			auto logical_index = columns.PhysicalToLogical(column_index);
+			constraints.push_back(make_uniq<NotNullConstraint>(logical_index));
+			bound_constraints.push_back(make_uniq<BoundNotNullConstraint>(column_index));
 		}
 	}
+
 	return bound_constraints;
 }
 
+unique_ptr<BoundConstraint> Binder::BindConstraint(Constraint &constraint, const string &table_name,
+                                                   const ColumnList &columns) {
+	switch (constraint.type) {
+	case ConstraintType::CHECK: {
+		unique_ptr<BoundConstraint> bound_constraint = make_uniq<BoundCheckConstraint>();
+		auto check_constraint = reinterpret_cast<BoundCheckConstraint *>(bound_constraint.get());
```

Comment:
`Cast<BoundCheckConstraint>`

Reply:
Will do this everywhere, not sure how I missed it.